core/arch/arm/kernel/kern.ld.S

```diff
@@ -364,22 +372,22 @@ SECTIONS
 				SMALL_PAGE_SIZE) * 32;
 	__tmp_hashes_end = __tmp_hashes_start + __tmp_hashes_size;
 
-	__init_mem_usage = __tmp_hashes_end - CFG_TEE_LOAD_ADDR;
+	__init_mem_usage = __tmp_hashes_end - TEE_TEXT_VA_START;
 
 	ASSERT(CFG_TEE_LOAD_ADDR >= CFG_TEE_RAM_START,
 		"Load address before start of physical memory")
 	ASSERT(CFG_TEE_LOAD_ADDR < (CFG_TEE_RAM_START + CFG_TEE_RAM_PH_SIZE),
 		"Load address after end of physical memory")
-	ASSERT(__tmp_hashes_end < (CFG_TEE_RAM_START + CFG_TEE_RAM_PH_SIZE),
+	ASSERT(__tmp_hashes_end < (TEE_RAM_VA_START + CFG_TEE_RAM_PH_SIZE),
 		"OP-TEE can't fit init part into available physical memory")
-	ASSERT((CFG_TEE_RAM_START + CFG_TEE_RAM_PH_SIZE - __init_end) >
+	ASSERT((TEE_RAM_VA_START + CFG_TEE_RAM_PH_SIZE - __init_end) >
```

Comment:
I prefer to original one: phys_addr + phys_size  rather than virt_addr + phys_size.
Yet there are equivalent.